linkerd/app/src/env.rs

```diff
@@ -448,7 +455,20 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
             } else {
                 outbound.proxy.connect.clone()
             };
+
+            // Add any extra Labels, that are not present in the labels file
+            let mut labels: HashMap<String, String> = [(
+                "namespace".to_string(),
+                namespace.unwrap_or_default().unwrap_or_default(),
+            )]
+            .iter()
+            .cloned()
+            .collect();
+
+            labels.extend(oc_trace_labels(oc_labels_file_path));
```

Comment:
IMO, creating a hashmap and then extending it with the labels from the file seems a little weird, although it does let us override the namespace label one from the file, if there is one. What about just doing
```suggestion
            let mut labels = oc_trace_labels(oc_labels_file_path);

            // If the file did not contain a `namespace` label, add one.
            labels.entry("namespace".to_string()).or_insert_with(|| namespace.flatten().unwrap_or_default());
            labels.extend(oc_trace_labels(oc_labels_file_path));
```